pghoard/rohmu/object_storage/swift.py

```diff
@@ -71,15 +74,36 @@ def list_path(self, key):
             })
         return return_list
 
+    def _delete_object_plain(self, key):
+        try:
+            return self.conn.delete_object(self.container_name, key)
+        except exceptions.ClientException as ex:
+            if ex.http_status == 404:
+                raise FileNotFoundFromStorageError(key)
+            raise
+
+    def _delete_object_segments(self, key, manifest):
+        self._delete_object_plain(key)
+        seg_container, seg_prefix = manifest.split("/", 1)
+        _, segments = self.conn.get_container(seg_container, prefix=seg_prefix, delimiter="/")
+        for item in segments:
+            if "name" in item:
+                with suppress(FileNotFoundFromStorageError):
+                    self._delete_object_plain(item["name"])
```

Comment:
you mean if it fails for any reason?  we already ignore 'file not found' errors so this shouldn't break if two instances start deleting the same multipart file simultaneously